src/org/traccar/reports/ReportUtils.java

```diff
@@ -250,100 +255,79 @@ private static boolean isMoving(ArrayList<Position> positions, int index,
         }
     }
 
-    public static Collection<BaseReport> detectTripsAndStops(TripsConfig tripsConfig, boolean ignoreOdometer,
-            double speedThreshold, Collection<Position> positionCollection, boolean trips) {
-
-        Collection<BaseReport> result = new ArrayList<>();
+    public static Collection<TripReport> detectTrips(Collection<Position> positionCollection, TripsConfig tripsConfig,
+            boolean ignoreOdometer, double speedThreshold) {
+        Collection<TripReport> result = new ArrayList<>();
 
         ArrayList<Position> positions = new ArrayList<>(positionCollection);
         if (positions != null && !positions.isEmpty()) {
-            int previousStartParkingIndex = 0;
+            MotionEventHandler  motionHandler = new MotionEventHandler(tripsConfig);
+            DeviceState deviceState = new DeviceState();
+            deviceState.setMotionState(isMoving(positions, 0, tripsConfig, speedThreshold));
+            int startTripIndex = deviceState.getMotionState() ? 0 : -1;
             int startParkingIndex = -1;
-            int previousEndParkingIndex = 0;
-            int endParkingIndex = 0;
-
-            boolean isMoving = false;
-            boolean isLast = false;
-            boolean skipped = false;
-            boolean tripFiltered = false;
-
             for (int i = 0; i < positions.size(); i++) {
-                isMoving = isMoving(positions, i, tripsConfig, speedThreshold);
-                isLast = i == positions.size() - 1;
-
-                if ((isMoving || isLast) && startParkingIndex != -1) {
-                    if (!skipped || previousEndParkingIndex == 0) {
-                        previousEndParkingIndex = endParkingIndex;
-                    }
-                    endParkingIndex = i;
+                positions.get(i).set(Position.KEY_MOTION, isMoving(positions, i, tripsConfig, speedThreshold));
```

Comment:
Maybe we can check if the attribute is already there and only calculate it if it's not?